### DISCUSSION TITLE
obtaining a tetrahedral msh file

### DISCUSSION THREAD

**Discussion by Ceciliaaasoldii** [sofa-framework/sofa#4691] [Ideas / Suggestions]
Good morning. i am trying to do a simulation in SOFA to move clips for intracranial aneurysms, obtained from CT images as .stl files
Through the use of programs like MeshMixer I was able to get the .obj files for the Visual model, however, I cannot get the .msh file to get the volumetric tetrahedral mesh for the mechanical model. I tried with Gmsh software but Visual Code reads the mesh obtained from the msh file as empty. i read that other people had the same problem as me and i tried MeshTetraStuffing. Unfortunately the links that have been shared cannot find the corresponding pages. if you could send me examples of application of this plugin I would be grateful. Thanks in advance

**Comment by epernod** [comment]
Hi @Ceciliaaasoldii 
I use Gmsh for that and it works fine if your stl file is clean (manifold triangulation) and closed. 
I usually export it as msh file in non binary format.
Then could you open the msh file you got and check it is not empty. If still not working could you post it here. 
Thanks
